### ISSUE TITLE
Split dataset layer into separate datasets

### ISSUE THREAD

**Issue by danielfdsilva** [NASA-IMPACT/veda-ui#493]
Currently the veda dashboard datasets have multiple layer. This is used as a logical grouping of data. Fox example, the [CO2 dataset](https://visex.netlify.app/air-quality/datasets/co2/explore) has 2 layers: Mean CO2 and Difference CO2

To be more inline with datasets on the STAC api and to lay ground for the explore page v2, the different layers should become their own datasets.

For the time being any dataset long form content will be duplicated when splitting the layers.

- [ ] Update veda-config documentation

**Comment by danielfdsilva** [comment]
This is also a good opportunity to add more properties to the dataset.
To start let's focus on the properties outlines on the [google doc](https://docs.google.com/document/d/1UT1bK7wTivPvSsTGP5073PISwYjbSsSTrhg6NizmnLM/edit#heading=h.y1ydzh1dcgo8)

**Comment by danielfdsilva** [comment]
The current dataset structure no longer fits the needs of the app. The new A&E page requires each dataset layer to be handled individually. This means we need to find a way to access and use each one of the layers separately.

Nevertheless, there should be a way to [group relevant layers together](https://github.com/NASA-IMPACT/veda-architecture/issues/277), ideally in a many-to-many relationship.

## Datasets

I propose that we extract each layer into a separate file and these become the new VEDA datasets. The veda dataset metadata will be added to each one of these files, meaning that in the beginning we will have a lot of duplicated metadata, like images, description and content. This is not ideal, but it is a necessary step to make the app work.

With the future in mind the data in these files will serve as an override/complement for the data in STAC. This means that when we open a dataset page we need to get the relevant data from STAC and merge it with the local data.

Each of these datasets will have an overview page with dedicated information, a way to open this dataset in the A&E page and a list of groups this dataset belongs to.

## Dataset Groups

The dataset groups will be a new content type added to the app. They will also have an overview page where content about the dataset group can be displayed together with a list of datasets that belong to this group. From this page, individual datasets can be explored or the whole group can be opened in the A&E page.

Both the dataset groups will appear in the data-catalog page to be searched and filtered.

## Design Considerations

With the creation of a new content (dataset groups) type we need a new single page and a new card. They can probably be very similar to the dataset page and card, but with some sort of differentiating feature.

Thoughts?

**Comment by nerik** [comment]
>  in the beginning we will have a lot of duplicated metadata, like images, description and content. This is not ideal, but it is a necessary step to make the app work.

If I understand correctly, this would be addressed with Dataset Groups, where images, description and content would go. Hence, I don't really understand the need for:

> With the creation of a new content (dataset groups) type we need a new single page and a new card. They can probably be very similar to the dataset page and card, but with some sort of differentiating feature. 

If this refacto is something we need to do as part of the A&E merge, why not go all the way with the following:
- Rename `Dataset`s to `DatasetGroup`s. 
- `DatasetGroup` retains all of former `Dataset` properties, except for
  - `layers` is removed
  - `datasetIds`, array of string, is introduced
- `DatasetLayer` is added as a content type, each one containing a former `layer` array element (formerly part of datasets)

**Comment by danielfdsilva** [comment]
That approach implies that each dataset layer must belong to at least 1 group (since the data catalog only shows dataset groups). This could lead to a situation where a standalone dataset needs a group just for itself to be able to be displayed. It seemed to me that we wanted to go more towards the approach that planetary computer takes where dataset layers and dataset groups are shown together. You can have datasets show up alone or as part of a group that makes sense.

I agree with the steps you listed, I'd just rename `DatasetLayer` to `Dataset` as to now mix the content with how they're displayed.

**Comment by nerik** [comment]
> This could lead to a situation where a standalone dataset needs a group just for itself to be able to be displayed.

Not necessarily; this is a UI choice, independent of the underlying data structure. For example, in the catalog, we can load groups and datasets, show all groups + datasets that don't belong to a group. If a specific page needs to be created to show a single dataset (not a group), we can also do that (it won't show any description, media, etc)

> I agree with the steps you listed, I'd just rename DatasetLayer to Dataset as to now mix the content with how they're displayed.

Completely with you on having UI language matching the system's entities name when possible. However, I'm a bit worried about refactoring if we have a new thing with the same name as an old, different one. But

**Comment by danielfdsilva** [comment]
That's definitely an option. We need to figure out if we want to have "empty" datasets pages. Would it be ok to have the various dataset individual meta details (from stac for example) on a group page or a dedicated page? I think this a matter both for the designer and for the scientists use case (cc @j08lue).

Personally I favor an option where datasets are more important and the groups are a secondary feature, but definitely not set on stone

**Comment by j08lue** [comment]
Good question how to present groups / single datasets.

I think users should not have to decide whether they need a group or a dataset, so we should not have a separate page in the frontend, but somehow flatten the experience between single datasets and groups.

For comparison, the Planetary Computer [catalog](https://planetarycomputer.microsoft.com/catalog) has three different types of catalog items side-by side - single datasets, groups, and "storage" datasets (those that are not explorable in the application because not cloud-optimized / in servable storage) - all of which are organized loosely by thematic area (Air Quality, Biodiversity, etc.).

Not sure what the heuristics are to generate these groups (see this [code for groups](https://github.com/microsoft/PlanetaryComputerDataCatalog/blob/66f93b60a17bb286df6561522132fb5bd566ec03/src/pages/CatalogGroup.js#L47C79-L59), I guess), but it seems to be dynamic and mostly relying on STAC (although probably not for the group metadata?).

We briefly touched upon the caveats of hosting data in STAC only and you mentioned that override of that content in Dashboard config has in the past been a requirement. I think we should revisit that requirement and see whether it is not outweighed by the needs for a single source of truth and more emphasis on STAC metadata for interoperability.

Anyways - those are future questions, although related... Hope to get to discuss this with you soon.

**Comment by nerik** [comment]
Related: https://github.com/NASA-IMPACT/veda-ui/pull/648

**Comment by aboydnw** [comment]
Closing due to inactivity
